File: PJS5_POC_Cpp/Sources/Weapon/MeleeWeapon/MeleeWeapon.cpp
#include "MeleeWeapon.h"

namespace WEAPON
{
	MELEE_WEAPON::MELEE_WEAPON(const int iId, const std::string& sName, const int iArmor, const int iLifePoint, const int iDamage, const int iPowerConsumption, const int iAccuracy, const int iMinAccuracy)
		: IWEAPON::IWEAPON(iId, sName, iArmor, iLifePoint, iDamage, iPowerConsumption, iAccuracy, iMinAccuracy, WEAPON_TYPE::MELEE_WEAPON)
	{
	}
}

File: PJS5_CSharp_Form/Sources/Weapon/MeleeWeapon/MeleeWeapon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WEAPON;

namespace PJS5_CSharp.Sources.Weapon.MeleeWeapon
{
    public class MELEE_WEAPON : IWeapon
    {
        public MELEE_WEAPON(int iId, string sName, int iArmor, int iLifePoint, int iDamage, int iPowerConsumption, int iAccuracy, int iMinAccuracy)
            : base(iId, sName, iArmor, iLifePoint, iDamage, iPowerConsumption, iAccuracy, iMinAccuracy, WEAPON_TYPE.MELEE_WEAPON)
        {
        }
    }
}
